pandas/core/arrays/_mixins.py

```diff
@@ -240,3 +241,11 @@ def _reduce(self, name: str, skipna: bool = True, **kwargs):
         else:
             msg = f"'{type(self).__name__}' does not implement reduction '{name}'"
             raise TypeError(msg)
+
+
+class StringArrayBase(ExtensionArray):
```

Comment:
This was for consistency with StringDtypeBase below and a first step towards https://github.com/pandas-dev/pandas/issues/35169#issuecomment-712372349

not actually needed for task in hand, i.e. fix astype failures.